src/Util/Serializer.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Stev\BTIPay\Util;
```

Comment:
@gabrielsolomon , ce ai creat tu aici e un factory. Cred ca e ok sa adaugam namespace-ul Stev\BTIPay\Factory si aici sa fie SerializerFactory. Ce parere ai ?
E ok ca l-ai facut si singleton.

Reply:
Ce parere ai avea de a face un namespace pe Serialize?
Ar putea sa exista un factory si un manager care sa faca munca.
Lucrul asta ar permite ca apoi sa poti schimba jms cu symfony serializer care are mai putin dependinte si stie si de annotations fara pachetul de la doctrine

Reply:
E ok un namespace Stev\BTIPay\Serializer si o clasa redenumita SerialiazerFactory sau Factory. 

Un manager nu cred ca e ok. Cel mult daca vrei tu poti sa scrii un SerializerWrapper sau SerialiazerDecorator ca asta faci de fapt. Ca din ce inteleg doar vei abstractiza apelul catre functiile de serialize si deserialize.
Dar chiar si asa nu vad beneficiul acestei clase suplimentare intrucat ambele serializatoare JMS si Symfony implementeaza o interfata cu aceleasi specificatii doar ca din pachete diferite.

M-as rezuma doar la Factory cu implementarea curenta de JMS. Schimbarea din JMS in Symfony va fi foarte simpla, intrucat vom schimba o singura clasa (Factory) si vom folosi o alta interfata dar care e identica cu cea din JMS.

Reply:
Am urcat acum pe namespace Serializer 

Eu ma gandeam la varianta in care fiind similare poti sa faci un discovery si sa foloseasca ori unul ori altul ca sa fie mai putin dependinte.

Reply:
Nu stiu daca discovery e ok. Serializatorul e o dependinta directa a librariei si e bine sa fie trecuta in composer.json
Si apare problema adnotarilor. Daca permiti sa fie ori JMS ori Symfony, cum pui adnotarile pt serializare ?
Simt ca e foarte multa complexitate pentru o problema inexistenta la momentul actual.

Daca ai o propunere concreta si timp, fa o implementare sa vedem cum arata.